src/lib/php/Dao/CopyrightDao.php

```diff
@@ -641,4 +655,59 @@ public function getTotalCopyrights($upload_pk, $item, $uploadTreeTableName, $age
     $iTotalRecordsRow = $this->dbManager->getSingleRow($countAllQuery, $params, __METHOD__, $tableName . "count.all" . ($activated ? '' : '_deactivated'));
     return $iTotalRecordsRow['count'];
   }
+
+  /**
+   * @brief get user copyright findings for a uploadtree
+   * @param int     $upload_pk           Upload id to get results from
+   * @param int     $item                Upload tree id of the item
+   * @param string  $uploadTreeTableName Upload tree table to use
+   * @param string  $type                Type of the statement (statement, url, email, author or ecc)
+   * @param boolean $activated           True to get activated copyrights, else false
+   * @return array[][] Array of table records, total records
+   */
+  public function getUserCopyrights($upload_pk, $item, $uploadTreeTableName, $type, $activated = true, $offset = null , $limit = null)
```

Comment:
This function looks very similar to `getCopyrights()` defined in `src/spasht/ui/ajax-spasht-copyright-hist.php`. If so, can you please remove the function `getUserCopyrights()` from the ajax file and call this new DAO function instead?

Reply:
The count of copyrights in UI and REST API differs, meaning difference in SQL queries.

![image](https://github.com/fossology/fossology/assets/18077542/e1f2a981-1a67-4d93-89f4-7e59cbbf35df)

Reply:
> This function looks very similar to `getCopyrights()` defined in `src/spasht/ui/ajax-spasht-copyright-hist.php`. If so, can you please remove the function `getUserCopyrights()` from the ajax file and call this new DAO function instead?

It's feasible, but I don't believe it would be the best option. The `getCopyrights()` function defined in `ajax-spasht-copyright-hist.php` accepts and utilises `agentId` variable which is a key requirement for `copyright_spasht` table (and all other related tables except `copyright_decision` which has `user_fk`). My sole purpose of writing a new function for getting user copyrights was to separate this different table structures. What do you think on this?

> The count of copyrights in UI and REST API differs, meaning difference in SQL queries.
> 
> ![image](https://private-user-images.githubusercontent.com/18077542/346047099-e1f2a981-1a67-4d93-89f4-7e59cbbf35df.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1OTY5NzIsIm5iZiI6MTcyMDU5NjY3MiwicGF0aCI6Ii8xODA3NzU0Mi8zNDYwNDcwOTktZTFmMmE5ODEtMWE2Ny00ZDkzLTg5ZjQtN2U1OWNiYmYzNWRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEwVDA3MzExMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkOWI2ZDUzZTY1OGYwZGFlYzg5ZGE1YjY3MjNjNjU1NzQzMTVhNTc3NzNiMDU4YjkwZjE3OWU3M2U5Mzg2MWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.w8Gs29RZWbW38At-8cOzkgvWbPMITDD_Lufj4A7vL2w)

Oh, that should not happen. I've pushed a fix that may address this; please inform me if the issue persists.

Reply:
Okay, makes sense. Anyways the ajax file will have to go once we integrate the new React UI. Also, the counts are correct now.